addons/web_editor/controllers/main.py

```diff
@@ -26,6 +27,47 @@
 logger = logging.getLogger(__name__)
 DEFAULT_LIBRARY_ENDPOINT = 'https://media-api.odoo.com'
 
+def generate_channel_signatures(env, model_name, field_name, res_id):
+    # Verify access to the edition channel.
+    if not env.user.has_group('base.group_user'):
+        raise AccessDenied()
+
+    rec = env[model_name].browse(res_id)
+
+    rec.check_access_rights('read')
+    rec.check_field_access_rights('read', [field_name])
+    rec.check_access_rule('read')
+    rec.check_access_rights('write')
+    rec.check_field_access_rights('write', [field_name])
+    rec.check_access_rule('write')
+
+    # expire in 1 hour
+    hour_timestamp = int(time.time() // 3600)
+
+    # Generate unique channel id as HMAC
+    # Looks like this: 3e1ef9e86...3036d65 (64 chars)
+    # timestamp is only valid 1 hour, then you have to renew it.
+    def get_signature(expiration_timestamp):
+        return odoo.tools.hmac(
+            env(su=True),
+            'editor_serialization',
+            f"{model_name}:{field_name}:{res_id}:{expiration_timestamp}"
+        )
+
+    # Return the expiration timestamp of the previous channel in hour and the
+    # previous, current and next signature.
+    return {
+        'next_expiration_timestamp': hour_timestamp + 1,
```

Comment:
Numeric timestamps usually implies seconds, even if this is rounded to the hour, would be less weird to stick to that.